src/main/java/frc/robot/subsystems/ShooterSubsystem

```diff
@@ -0,0 +1,119 @@
+
+    //change later
+    int IDNUMBER = 10; //so I remember to change them later
+    double speed = 0.1; 
+    int final ERRORTOLERANCE = 10;
+    int final LIMIT_SWITCH_ID = 1;
+    int final GEARBOX_RATIO = 1/20;
+
+    //angle PID (CHANGE LATER)
+    private static final double angleP = 2.4;
+    private static final double angleI = 0;
+    private static final double angleD = 0;
+
+    //motors
+    private final WPI_TalonSRX conveyerBelt; 
+    private final CANSparkMax pivotMotor;
+    private final CANSparkMax shooterMotor;
+
+    //devices
+    private RelativeEncoder rotationEncoder;
+    private SparkMaxPIDController rotationPidController;
+    private final DigitalInput limitSwitch;
+    private robotState currentState;
+
+    //sensors
+    private final AnalogPotentiometer shooterSensor; //distance sensor
+
+    public ShooterSubsystem(){
+        //motors
+        conveyerBelt = new WPI_TalonSRX(10);
+        pivotMotor = new CANSparkMax(10, MotorType.kBrushed);
+        shooterMotor = new CANSparkMax(10, MotorType.kBrushed);
+
+        //devices
+        rotationEncoder = pivotMotor.getEncoder();
+        rotationPIDController = pivotMotor.getPIDController();
+        limitSwitch = new DigitalInput(LIMIT_SWITCH_ID);
+        
+        //setting PID vars
+        rotationPIDController.setP(extensionP);
+        rotationPIDController.setI(extensionI);
+        rotationPIDController.setD(extensionD);
+
+        //encoder stuff
+        rotationEncoder.setPositionConversionFactor(GEARBOX_RATIO);
+
+        //enums
+        state = robotState.NO_NOTE;
+
+        rotationPIDController.setSmartMotionAllowedClosedLoopError(ERRORTOLERANCE, 0); //what does 0 do (slotID is from 0-3)
+        setShooterState(robotState.NO_NOTE);
+    }
+
+    //enum functions
+    public robotState getShooterState(robotState state){
+        return currentState;
+    }
+
+    public robotState setShooterState(robotState newState){
+        currentState = newState;
+    }
+
+    public shootNote(){
+        if(shooterSensor.get() < tolerance){
+            conveyerBelt.set(speed);
+            setShooterState(robotState.FIRING);
+        } else {
+            shooterMotor.set(0);
+            conveyerBelt.set(speed);
+        }
+    }
+
+    public setAngle(double angle){
+        rotationEncoder.setPosition(angle);
+    }
+
+    public void periodic(){
+        //resets relative encoder every time robot starts again
+        if(limitSwitch != null && !limitSwitch.get()){ //false = limit switch is pressed
+            rotationEncoder.setPosition(0); 
+        }
+
+        if(currentState == FIRING && (shooterSensor.get() < tolerance)){  //when there is no note
+            setShooterState(robotState.NO_NOTE);
+        }
+
+        // switch(currentState) {
+        //     case HOLDING_NOTE:
+        //       shooterMotor.set(speed); //warming up the motor
+        //       break;
+        //     case LOADING_NOTE:
+        //        shooterMotor.set(speed);
```

Comment:
We can try doing this in testing, but this should probably be voltage. Setting a voltage ensures that the same speed is set consistently. The regular set function does a percent out, which is multiplied by the current battery voltage to set the speed. the voltage will fluctuate between 13.5 and 11.5 which may cause very different speeds of the flywheel over the course of a match and between different matches.